src/Repository/SearchInputRepository.php

```diff
@@ -0,0 +1,358 @@
+<?php
+/**
+ * Clarifai SearchInput
+ *
+ * @category Input
+ * @package  Clarifai
+ * @author   Darryn Ten <darrynten@github.com>
+ * @license  MIT <https://github.com/darrynten/clarifai-php/LICENSE>
+ * @link     https://github.com/darrynten/clarifai-php
+ */
+
+namespace DarrynTen\Clarifai\Repository;
+
+use DarrynTen\Clarifai\Entity\Concept;
+use DarrynTen\Clarifai\Entity\Input;
+
+/**
+ * Single Clarifai Input
+ *
+ * @package Clarifai
+ */
+class SearchInputRepository extends InputRepository
+{
+    /**
+     * parameter for input Concepts search
+     */
+    const INPUT_CONCEPTS = 'input_concepts';
+
+    /**
+     * parameter for output Concepts search
+     */
+    const OUTPUT_CONCEPTS = 'output_concepts';
+
+    /**
+     * parameter for input Metadata search
+     */
+    const METADATA = 'metadata';
+
+    /**
+     * parameter for input image search
+     */
+    const IMAGES = 'images';
+
+    /**
+     * parameter for reverse image search
+     */
+    const REVERSED_IMAGES = 'reversed_images';
+
+    /**
+     * Searches Inputs
+     *
+     * @param array $params
+     *
+     * @return Input[] array
+     */
+    public function search($params)
+    {
+        $searchResult = $this->getRequest()->request(
+            'POST',
+            'searches',
+            $this->getSearchQuery($params)
+        );
+
+        return $this->getInputsFromSearchResult($searchResult);
+    }
+
+    /**
+     * @param array $params
+     *
+     * @return array
+     */
+    public function getSearchQuery($params)
+    {
+        $data['query'] = [];
+        $data['query']['ands'] = [];
+
+        if (isset($params[self::INPUT_CONCEPTS])) {
+            $data['query']['ands'] = $this->generateInputConceptsQuery(
+                $data['query']['ands'],
+                $params[self::INPUT_CONCEPTS]
+            );
+        }
+
+        if (isset($params[self::OUTPUT_CONCEPTS])) {
+            $data['query']['ands'] = $this->generateOutputConceptsQuery(
+                $data['query']['ands'],
+                $params[self::OUTPUT_CONCEPTS]
+            );
+        }
+
+        if (isset($params[self::METADATA])) {
+            $data['query']['ands'] = $this->generateMetadataQuery($data['query']['ands'], $params[self::METADATA]);
+        }
+
+        if (isset($params[self::IMAGES])) {
+            $data['query']['ands'] = $this->generateImagesQuery($data['query']['ands'], $params[self::IMAGES]);
+        }
+
+        if (isset($params[self::REVERSED_IMAGES])) {
+            $data['query']['ands'] = $this->generateReverseImagesQuery(
+                $data['query']['ands'],
+                $params[self::REVERSED_IMAGES]
+            );
+        }
+
+        return $data;
+    }
+
+    /**
+     * Generates Input Concept search query and adds it to existing data
+     *
+     * @param $data
+     * @param Concept[] $concepts
+     *
+     * @return array $data
+     */
+    public function generateInputConceptsQuery($data, $concepts)
```

Comment:
This method and `generateOutputConceptsQuery` looks the same, split the same logic into different method